hs/src/Reach/Simulator/Core.hs

```diff
@@ -326,11 +330,11 @@ addToStore x v = do
       let lst' = lst {l_store = M.insert x v st}
       setLocal $ l {l_locals = M.insert aid lst' locals}
 
-fixMessageInRecord :: PhaseId -> ActorId -> Store -> DLPayAmt -> App ()
-fixMessageInRecord phId actId sto pay = do
+fixMessageInRecord :: PhaseId -> ActorId -> Store -> DLPayAmt -> Bool -> App ()
+fixMessageInRecord phId actId sto pay apiFlag = do
   (g, _) <- getState
   let msgs = e_messages g
-  let msgs' = M.insert phId (Fixed (actId, Message {m_store = sto, m_pay = pay})) msgs
+  let msgs' = M.insert phId (Fixed (actId, Message {m_store = sto, m_pay = pay, m_api = apiFlag})) msgs
```

Comment:
Pro-tip: Rename the variables `m_store`, `m_pay`, and `m_pay`, then write `Message {..}`